### DISCUSSION TITLE
Possible to avoid scroll-to-top when opening the dialog?

### DISCUSSION THREAD

**Discussion by searls** [josefarias/hotwire_combobox#186] [Q&A]
Try as I might, whenever I'm in mobile and [showModal() is called](https://github.com/josefarias/hotwire_combobox/blob/main/app/assets/javascripts/hw_combobox/models/combobox/toggle.js#L123) the page is scrolled to the top as the dialog comes up and then scrolled back down when the dialog is dismissed. 

This is pretty jarring, especially if you set `--hw-dialog-top-offset` to a larger value so that the dialog doesn't take up the entire screen (I've had customers complain that they get thrown off by how it takes over the screen).

Any ideas?

**Comment by josefarias** [comment]
Hm, I think this may come from the vendored library we're using to lock body scrolling.

I've been meaning to replace that with our own implementation which hopefully can sidestep this.

Locking body scrolling when a modal is open is particularly hard in mobile Safari. So let's hope it's not a hard requirement.

**Reply by searls** [reply]
For what it's worth, I tried commenting out the body scroll lock / unlock calls and (while it indeed allowed scrolling and broke the UI), the modal zipped to the top of the page anyway. Googling around for the issue, it seems like people most often call out input focus as the culprit (I wonder, but haven't tested, if the timing of focus()'ing on the dialog input has something to do with it)

**Comment by josefarias** [comment]
@searls This should be fixed now in https://github.com/josefarias/hotwire_combobox/pull/215. If you get a moment, could I get you to try out that branch and validate the fix? Thanks!

**Reply by josefarias** [reply]
Actually, you might wanna hold off. I'll try to solve https://github.com/josefarias/hotwire_combobox/pull/215#discussion_r1798649243 before merging.

**Reply by josefarias** [reply]
Alright @searls I've merged that. If you get a moment, could I get you to try it out and see if it behaves as expected now? 🙏

Closing for now but feel free to reopen. Thanks!
